lib/vehicle_factory.rb

```diff
@@ -0,0 +1,16 @@
+class VehicleFactory
+  attr_reader
+
+  def create_wa_vehicle(vehicle)
+    vehicle = vehicle.slice(:vin_1_10, :model, :make, :model_year).merge({engine: :ev})
+    vehicle[:year] = vehicle.delete(:model_year)
+    vehicle[:vin] = vehicle.delete(:vin_1_10)
+    Vehicle.new(vehicle)
+  end
+
+  def create_vehicles(registrations)
+    registrations.map do |registration|
+      self.create_wa_vehicle(registration)
+    end
```

Comment:
This is good, nice use of the map. Again though, we could be storing this array in an instance variable so everytime we want these vehicles, we don't have to do all this mapping everytime. Also, we don't need the `self` keyword on line 13